### PR TITLE
Added DateTimeGenerators and LocalDateTimeGenerators.

### PR DESCRIPTION
**Pull request by jeffmay**
@chrissalij @kylemessner
- Generators for joda-time classes
- Minor cleanup

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -25,9 +25,13 @@ scalacOptions := {
...
 
 libraryDependencies := Seq(
+  "org.joda" % "joda-convert" % "1.7",
+  "joda-time" % "joda-time" % "2.8",
   "org.scalacheck" %% "scalacheck" % "1.12.2",
   "org.scalatest" %% "scalatest" % "2.2.4"
-).map(_.withSources())
+)
+
+testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-oF")
```
Comment (ghost) [inline build.sbt:34]: What does this do?

File: build.sbt [hunk 2]
```diff
@@ -25,9 +25,13 @@ scalacOptions := {
 )
 
 libraryDependencies := Seq(
+  "org.joda" % "joda-convert" % "1.7",
```
Comment (ghost) [inline build.sbt:28]: What is joda-convert needed for?

File: build.sbt (hunk 1 above)
Comment (jeffmay) [inline build.sbt:34]: Shows full stack traces in test failures.

File: build.sbt (hunk 2 above)
Comment (jeffmay) [inline build.sbt:28]: I was getting compiler errors about missing stubs. I think it is a library that joda-time integrates with that for some reason is causing runtime errors when looking for converters of things. I'll try to remove this, but it seems harmless.

Comment (jeffmay) [inline build.sbt:28]: Without it, I get the following compiler errors:

```
[info] Compiling 17 Scala sources to /code/scalacheck-ops/target/scala-2.10/classes...
[error] Class org.joda.convert.FromString not found - continuing with a stub.
[error] Class org.joda.convert.ToString not found - continuing with a stub.
[error] Class org.joda.convert.FromString not found - continuing with a stub.
[error] Class org.joda.convert.FromString not found - continuing with a stub.
[error] Class org.joda.convert.ToString not found - continuing with a stub.
[error] Class org.joda.convert.ToString not found - continuing with a stub.
[error] Class org.joda.convert.ToString not found - continuing with a stub.
[error] Class org.joda.convert.FromString not found - continuing with a stub.
[error] Class org.joda.convert.ToString not found - continuing with a stub.
[error] 9 errors found
[error] (compile:compileIncremental) Compilation failed
[error] Total time: 5 s, completed May 31, 2015 3:38:10 PM
```

**Comment by ghost** [comment]
LGTM

File: src/main/scala/org/scalacheck/ops/time/joda/DateTimeGenerators.scala [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    try datetime plus duration
+    catch {
+      case tooLarge: ArithmeticException => params.chronology.maxDateTime
+    }
+  }
+
+  override protected[time] def subtractToFloor(datetime: DateTimeType, duration: DurationType)(implicit params: ParamsType): DateTimeType = {
+    try datetime minus duration
+    catch {
+      case tooLarge: ArithmeticException => new DateTime(0, params.chronology)
```
Comment (jeffmay) [inline src/main/scala/org/scalacheck/ops/time/joda/DateTimeGenerators.scala]: This should be `tooSmall` and why not have a `ChronologyOps.minDateTime`?

File: src/main/scala/org/scalacheck/ops/time/joda/ImplicitJodaTimeGenerators.scala [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  implicit val arbDateTimeZone: Arbitrary[DateTimeZone] = {
+    val ids = DateTimeZone.getAvailableIDs.toSeq
+    val zones = ids map DateTimeZone.forID
+    Arbitrary(oneOf(zones))
+  }
+
+  implicit def arbChronology(implicit arbZone: Arbitrary[DateTimeZone]): Arbitrary[Chronology] = {
+    val chronologyBuilders: Seq[DateTimeZone => Chronology] = Seq(
+      BuddhistChronology.getInstance,
+      CopticChronology.getInstance,
```
Comment (jeffmay) [inline src/main/scala/org/scalacheck/ops/time/joda/ImplicitJodaTimeGenerators.scala:21]: Rally Egypt!

Also, I think we support i18n for ⲙⲉⲧⲣⲉⲙ̀ⲛⲭⲏⲙⲓ too, right?

Comment (jeffmay) [inline src/main/scala/org/scalacheck/ops/time/joda/ImplicitJodaTimeGenerators.scala:21]: Gosh, Google Chrome, get with the times! :trollface:

File: src/main/scala/org/scalacheck/ops/time/joda/LocalDateTimeGenerators.scala [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    catch {
+      case math: ArithmeticException => params.chronology.maxLocalDateTime
+    }
+  }
+
+  override protected[time] def subtractToFloor(datetime: LocalDateTime, duration: ReadableDuration)
+    (implicit params: JodaTimeParams): LocalDateTime = {
+    try datetime minus duration
+    catch {
+      case math: ArithmeticException => new LocalDateTime(0, params.chronology)
```
Comment (jeffmay) [inline src/main/scala/org/scalacheck/ops/time/joda/LocalDateTimeGenerators.scala]: MATHS!

**Comment by KyleMessner** [comment]
LGTM

File: build.sbt (hunk 2 above)
Comment (slustbader) [inline build.sbt:28]: http://www.joda.org/joda-time/upgradeto230.html: 

> Joda-Time uses annotations from Joda-Convert.
> In the Java programming language, this dependency is optional, however in Scala it is not.
> Scala users must manually add the Joda-Convert v1.2 dependency.

**Pull request merged**